torchci/clickhouse_queries/commit_jobs_query/query.sql

```diff
@@ -1,158 +1,117 @@
+        INNER JOIN workflow_run workflow final ON workflow.id = job.run_id
+    WHERE
+        job.name != 'ciflow_should_run'
+        AND job.name != 'generate-test-matrix'
+        AND workflow.event != 'workflow_run' -- Filter out workflow_run-triggered jobs, which have nothing to do with the SHA
+        AND workflow.event != 'repository_dispatch' -- Filter out repository_dispatch-triggered jobs, which have nothing to do with the SHA
+        AND workflow.head_sha = {sha: String }
+        AND job.head_sha = {sha: String }
+        AND workflow.repository. 'full_name' = {repo: String } --         UNION
+    UNION ALL
+    SELECT
+        workflow.created_at AS time,
+        workflow.head_sha AS sha,
+        workflow.name AS job_name,
+        'Workflow Startup Failure' AS workflow_name,
+        workflow.id,
+        0 AS workflow_id,
+        workflow.artifacts_url AS github_artifact_url,
+        if(
+            workflow.conclusion = ''
+            and workflow.status = 'queued',
+            'failure',
+            workflow.conclusion
+        ) as conclusion,
+        workflow.html_url,
+        '' AS log_url,
+        DATE_DIFF(
+            'SECOND',
+            workflow.created_at,
+            workflow.run_started_at
+        ) AS queue_time_s,
+        0 AS duration_s,
+        '' as line,
+        [ ] as captures,
+        0 as line_num,
+        [ ] as context,
+        '' AS runner_name,
+        workflow.head_commit.author.email AS authorEmail
+    FROM
+        workflow_run workflow final
+    WHERE
+        workflow.event != 'workflow_run' -- Filter out workflow_run-triggered jobs, which have nothing to do with the SHA
+        AND workflow.event != 'repository_dispatch' -- Filter out repository_dispatch-triggered jobs, which have nothing to do with the SHA
+        AND workflow.head_sha = {sha: String }
+        AND workflow.repository.full_name = {repo: String }
 )
 SELECT
-  sha,
-  workflow_name AS workflowName,
-  job_name AS jobName,
-  CONCAT(workflow_name, ' / ', job_name) AS name,
-  id AS id,
-  workflow_id AS workflowId,
-  github_artifact_url AS githubArtifactUrl,
-  if(
-    conclusion = '', 'pending', conclusion
-  ),
-  html_url AS htmlUrl,
-  log_url AS logUrl,
-  duration_s AS durationS,
-  queue_time_s AS queueTimeS,
-  line AS failureLines,
-  line_num AS failureLineNumbers,
-  captures AS failureCaptures,
-  context AS failureContext,
-  runner_name AS runnerName,
-  authorEmail,
-  time,
+    sha,
+    workflow_name AS workflowName,
+    job_name AS jobName,
+    CONCAT(workflow_name, ' / ', job_name) AS name,
+    id AS id,
+    workflow_id AS workflowId,
+    github_artifact_url AS githubArtifactUrl,
+    if(conclusion = '', 'pending', conclusion) as conclusion,
+    html_url AS htmlUrl,
+    log_url AS logUrl,
+    duration_s AS durationS,
+    queue_time_s AS queueTimeS,
+    if(line = '', [ ], [ line ]) AS failureLines,
+    if(line_num = 0, [ ], [ line_num ]) AS failureLineNumbers,
```

Comment:
Could you please explain what this bit is doing?

Reply:
Converts into arrays because that is the format the current code is expected.

Added comment in code